File: Javascript/Algorithms/1684.js
let countConsistentStrings = (allowed, words) => {
    const allowed_set = new Set(allowed);
    console.log(allowed_set);
    let count = 0, flag = 1;
    for (let w of words){
        for (let c of w){
            if (!allowed_set.has(c)) {
                flag = 0;
                break;
            }
        }
        if (!flag){
            flag = 1;
            continue;
        }
        count++;
    }
    return count;
};

File: C/Algorithms/1684.c
#include "uthash/src/uthash.h"

typedef struct {
  int value;
  UT_hash_handle hh;
} s_entry;

int countConsistentStrings(char *allowed, char **words, int wordsSize) {
  // create set of allowed characters.
  s_entry *allowed_set = NULL;
  while (*allowed) {
    s_entry *e = malloc(sizeof(*e));
    e->value = *allowed++;
    HASH_ADD_INT(allowed_set, value, e);
  }

  int count = 0, flag = 1;
  for (int i = 0; i < wordsSize; i++) {
    char *s = words[i];
    while (*s) {
      s_entry *e;
      int value = *s++;
      HASH_FIND_INT(allowed_set, &value, e);
      if (!e) {
        flag = 0;
        break;
      }
    }
    if (!flag) {
      flag = 1;
      continue;
    }
    count++;
  }
  HASH_CLEAR(hh, allowed_set);
  return count;
}
